### DISCUSSION TITLE
Failed to build strimzi-kafka-operator

### DISCUSSION THREAD

**Discussion by mqware** [strimzi/strimzi-kafka-operator#10257] [Q&A]
Hello,

I am trying to build [strimzi-kafka-operator](https://github.com/strimzi/strimzi-kafka-operator), so I cloned the repository and ran "mvn clean" as instructed in [Build and deploy Strimzi from source](https://github.com/strimzi/strimzi-kafka-operator/blob/main/development-docs/DEV_GUIDE.md#build-and-deploy-from-source), but I keep getting the following error:

```
[ERROR] Failed to execute goal on project config-model-generator: Could not resolve dependencies for project io.strimzi:config-model-generator:jar:0.42.0-SNAPSHOT
[ERROR] dependency: io.strimzi:config-model:jar:0.42.0-SNAPSHOT (compile)
[ERROR]         Could not find artifact io.strimzi:config-model:jar:0.42.0-SNAPSHOT
```

I tried it multiple times both on Windows and Linux, but always with the same result. At the end I was able to get around this by extracting the config-model jar from the container, and installing it into my local Maven repository, so Maven can find it, (and switching to the release-0.41.x branch to match the version of the jar), but that's obviously a hack. Can you help me figure out what am I doing wrong here? I am not familiar with Maven, but unlike other poms, the pom.xml for config-model doesn't have a jar section, so I don't understand how that jar would be built.

Thanks in advance.

**Comment by scholzj** [comment]
Just run something like `mvn install` if you just checked it out?

**Comment by mqware** [comment]
I tried `mvn install`, but that failed in test `io.strimzi.test.mockkube3.MockKube3ControllersMockTest.testDeploymentController`. I assume this is because something is not configured correctly in my environment. But `mvn install -DskipTests` worked fine. Thanks!

**Reply by scholzj** [reply]
That could be something like missing Docker or something like that. But hard to say without the logs.

**Reply by mqware** [reply]
I am not sure what it was, but I cleaned up my Docker environment and reset Microk8s, and then the above test passed. However, now I am stuck on this one;
```
[ERROR] Failures:
[ERROR]   MainIT.testCreateClusterRolesCreatesClusterRoles(VertxTestContext) io.vertx.core.impl.NoStackTraceThrowable: Failed to create Cluster Roles.
```
I attached the test result.
Thank you.

[TEST-io.strimzi.operator.cluster.MainIT.xml.zip](https://github.com/user-attachments/files/15971664/TEST-io.strimzi.operator.cluster.MainIT.xml.zip)

**Reply by scholzj** [reply]
Sounds like that expects your Kubernetes cluster to have a set default namespace which you don't have (e.g. in your `.kubeconfig` context):

```yaml
# ...
- context:
    cluster: my-kube:6443
    namespace: myproject # <=== THSI MIGHT BE MISSING IN YOUR CONFIG
    user: kube:admin/my-kube:6443
  name: my-kube
# ...
```

Could you please open a bug for it? https://github.com/strimzi/strimzi-kafka-operator/issues ... I will try to have a look into it.

**Reply by scholzj** [reply]
Actually, we do not use this feature anymore anyway. So I just removed it in #10265. So there's no need to open an issue anymore. Thanks.

**Reply by mqware** [reply]
I tried what you suggested, but unfortunately it didn't work. The first time I got this error:
```
io.fabric8.kubernetes.client.KubernetesClientException: Operation: [get]  for kind: [ClusterRole]  with name: [strimzi-kafka-broker]  in namespace: [null]  failed.
```
And now this:
```
io.fabric8.kubernetes.client.KubernetesClientException: Operation: [get]  for kind: [ClusterRole]  with name: [strimzi-kafka-broker]  in namespace: [myproject]  failed.
```
So, it looks like the namespace got picked up from the config, but the test still failed. Since you are removing this code anyway, I will just wait for that change.
Thanks.

**Reply by scholzj** [reply]
That is strange. In any case, I removed it in the PR I linked, so once it is merged it should be solved.
